artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ServerSessionImpl.java

```diff
@@ -153,7 +153,7 @@
 
    protected volatile boolean started = false;
 
-   protected final Map<SimpleString, TempQueueCleanerUpper> tempQueueCleannerUppers = new HashMap<>();
+   protected static final Map<SimpleString, TempQueueCleanerUpper> tempQueueCleannerUppers = new HashMap<>();
```

Comment:
this does not work well here. 

A HashMap on ServerSession is fine, as it represents a single thread context of the client.

however, if you make it static. .that's an issue.


Still, i am not sure i would make it static here, perhaps store such thing on the Server or other controller?

Reply:
What occurs where multiple embedded brokers in same jvm? 
Always super cautious of state in statics.

Reply:
As temp dest are tied to the connection should maybe the map live there and shared on all sessions on thay connection. Thus removing the static but also being correct with regards to being able to share between sessions on same connection

Reply:
@michaelandrepearce, that's good idea. I'll look into that...

Reply:
Also if being shared between sessions on a connection need to handle concurrency. Maybe simply use concurrenthashmap

Reply:
I have an alternative approach...

Reply:
I'm sending a PR now

Reply:
@jbertram I sent a PR to your branch

Reply:
The issue is that the session is temp is always creating a new session to delete the queue.

I'm reusing the original session on my patch, and only opening a new one if it happened to be closed. that's a minimal change here.

Reply:
I put comment on your pr to justins.

Essentially what about other protocols...

Reply:
@clebertsuconic, I'm using your change, but I squashed the commits. I hope that's ok.

@michaelandrepearce, I don't think there's going to be any issue on the other protocols because the JMS client should be the only one that was creating a new session every time when deleting a temporary queue.

Reply:
fair enough, i guess im just worried temp queues being fixed (the prev pr that caused this) for core all well and good, but we miss the other protocols and then find a more central solution for all would have been better...

anyhow +1 this for the immediate.